### PR TITLE
Final conversion of query builder from Joomla code to Hubzero

### PR DESCRIPTION
**Pull request by zooley**

### CODE REVIEW

File: src/Document/Type/Html.php [hunk 1]
```diff
@@ -431,12 +431,14 @@ public function countMenuChildren()
 			{
 				$dbo = \App::get('db');
 
-				$query = $dbo->getQuery(true);
-				$query->select('COUNT(*)');
```
Comment (kevinwojo) [inline src/Document/Type/Html.php]: This looks like it returns a count.

File: src/Document/Type/Html.php [hunk 2]
```diff
@@ -431,12 +431,14 @@ public function countMenuChildren()
...
-				$query->from('#__menu');
-				$query->where('parent_id = ' . $active->id);
-				$query->where('published = 1');
-
+				$query = $dbo->getQuery();
```
Comment (kevinwojo) [inline src/Document/Type/Html.php]: This does not appear to return a count.

Comment (zooley) [inline src/Document/Type/Html.php]: The query starts:

```
$query
    ->select('id', null, true)
    ...
```

The TRUE at the end there tells it to do a COUNT on the field.

Comment (kevinwojo) [inline src/Document/Type/Html.php]: TIL

**Comment by kevinwojo** [comment]
LGTM

**Pull request merged**